org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/indexing/AddFolderToIndex.java

```diff
@@ -103,8 +103,7 @@ public boolean visit(IResourceProxy proxy) /* throws CoreException */{
 			}
 		} catch (CoreException e) {
 			if (JobManager.VERBOSE) {
-				Util.verbose("-> failed to add " + this.folderPath + " to index because of the following exception:", System.err); //$NON-NLS-1$ //$NON-NLS-2$
-				e.printStackTrace();
+				Util.log(e, "-> failed to add " + this.folderPath + " to index because of the following exception:"); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
This is not OK, see my explanation above.

Reply:
If I understand you correctly, 
* All calls to `Util.verbose("...")` and `Util.verbose("...", System.err)` are **debug information** and they should be done via the `DebugTrace` (in another PR)
* All calls to `e.printStackTrace()` are meant to be logged and they should be replaced by a call to `Util.log(e)` in this PR.

?

Reply:
Almost correct.

> All calls to e.printStackTrace()

All calls to e.printStackTrace() (or System.(out|err).print* etc) in context of "usual" error handling where no "Verbose/Debug/Trace" etc code is involved and where we (still) have platform running (there are cases on shutdown where no bundle logging is possible).

Reply:
Thank you, I hope I got it right this time :)

Reply:
> Thank you, I hope I got it right this time :)

Not really, see below.